tmc-langs-framework/src/test/java/fi/helsinki/cs/tmc/langs/ExerciseBuilderTest.java

```diff
@@ -0,0 +1,118 @@
+        instance.prepareStub(targetFolder.toPath());
+       
+        File expectedFolder = new File("src/test/resources/arith_funcs_stub/src");
+        File outputFolder = new File(targetFolder.getAbsolutePath() + "/src");
+        
+        Map<String, File> expectedFiles = getFileMap(expectedFolder);
+        Map<String, File> actualFiles = getFileMap(outputFolder);
+        
+        for(String fileName : expectedFiles.keySet()) {
+            File expected = expectedFiles.get(fileName);
+            File actual = actualFiles.get(fileName);
+            List<String> expectedLines = FileUtils.readLines(expected);
+            List<String> actualLines = FileUtils.readLines(actual);
+            for (int i = 0 ; i < expectedLines.size() ; ++i) {
+                String expectedLine = expectedLines.get(i);
+                String actualLine = actualLines.get(i);
+                assertEquals("Line in file " + fileName + " did not match ", expectedLine, actualLine);
+            }
+            
+        }
+        
+    }
+
+    private Map<String, File> getFileMap(File folder) throws IOException {
+        if (!folder.isDirectory()) {
+            throw new IOException("Supplied path was not a directory");
+        }
+        Map<String, File> result = new HashMap<>();
+        for (File file : folder.listFiles()) {
+            if (file.isDirectory()) {
+                result.putAll(getFileMap(file));
+                continue;
+            }
+            result.put(file.getName(), file);
+        }
+        return result;
+    }
+
+    /**
+     * Test of prepareSolution method, of class ExerciseUtils.
+     */
+    @Test
+    public void testPrepareSolution() throws IOException {
+        System.out.println("prepareSolution");
+        File originProject = new File("src/test/resources/arith_funcs/");
+
+        File targetFolder = createTemporaryCopyOf(originProject);
+        ExerciseBuilder instance = new ExerciseBuilder();
+
+        instance.prepareSolution(targetFolder.toPath());
+       
+        File expectedFolder = new File("src/test/resources/arith_funcs_solution/src");
+        File outputFolder = new File(targetFolder.getAbsolutePath() + "/src");
+        
+        Map<String, File> expectedFiles = getFileMap(expectedFolder);
+        Map<String, File> actualFiles = getFileMap(outputFolder);
+        
+        for(String fileName : expectedFiles.keySet()) {
+            File expected = expectedFiles.get(fileName);
+            File actual = actualFiles.get(fileName);
+            List<String> expectedLines = FileUtils.readLines(expected);
+            List<String> actualLines = FileUtils.readLines(actual);
+            for (int i = 0 ; i < expectedLines.size() ; ++i) {
+                String expectedLine = expectedLines.get(i);
+                String actualLine = actualLines.get(i);
+                assertEquals("Line in file " + fileName + " did not match ", expectedLine, actualLine);
+            }
+            
+        }
```

Comment:
This directory comparison logic is repeated. Factor it out into a helper method.